src/compute-engine/library/arithmetic.ts

```diff
@@ -1395,7 +1392,7 @@ export const ARITHMETIC_LIBRARY: IdentifierDefinitions[] = [
         if (s === undefined) return undefined;
         if (['positive', 'zero'].includes(s)) return s;
         const n = reduceBigOp(ops[0], ops.slice(1), (acc, x) => acc + 1, 0);
-        if (s === 'real-not-zero' || s === 'negative')
+        if ((ops[0]?.type === 'real' && s === 'not-zero') || s === 'negative')
```

Comment:
Checking that type === 'real' is probably not correct (it could be 'integer' or 'rational' and this condition would presumably still apply). I think you meant to check ops[0]?.isReal, which will account for real or one of its subtype.

Reply:
It is my mistake for not checking possible values of type. I thought it was just real, imaginary, complex